Marlin/servo.cpp

```diff
@@ -242,11 +242,9 @@ uint8_t Servo::attach(int pin) {
 
 uint8_t Servo::attach(int pin, int min, int max) {
   if (this->servoIndex < MAX_SERVOS ) {
-  #if defined(ENABLE_AUTO_BED_LEVELING) && (PROBE_SERVO_DEACTIVATION_DELAY > 0)
-    if (pin > 0) this->pin = pin; else pin = this->pin;
-  #endif
-    pinMode(pin, OUTPUT);                                   // set servo pin to output
-    servos[this->servoIndex].Pin.nbr = pin;
+    if(pin > 0)
+      servos[this->servoIndex].Pin.nbr = pin;
+    pinMode(servos[this->servoIndex].Pin.nbr, OUTPUT); // set servo pin to output
```

Comment:
Yeah, thought about this one too. Much cleaner, and the code should be just as small.